src/gravel/controllers/kv.py

```diff
@@ -12,125 +12,395 @@
+                    # this can throw (auth failed, etc.)
+                    logger.debug("Connecting to cluster")
+                    self._cluster.connect()
+                    has_aquarium_pool = "aquarium" in self._cluster.list_pools()
+                    if not has_aquarium_pool:
+                        logger.info("Creating aquarium pool")
+                        # consider setting pg_num 1 as with device_health_metrics pool
+                        # also need to do the equivalent of
+                        # `ceph osd pool application enable aquarium aquarium`
+                        # (see devicehealth mgr module)
+                        self._cluster.create_pool("aquarium")
+                    self._ioctx = self._cluster.open_ioctx("aquarium")
+                    # This actually seems to be safe (doesn't trash existing omap
+                    # data if present, which is neat)
+                    self._ioctx.write_full(
+                        "kvstore",
+                        "# aquarium kv store is in this object's omap\n".encode(
+                            "utf-8"
+                        ),
+                    )
+                    # At this point, iff it's a new pool, new object, etc.
+                    # we need to push everything from our local cache to
+                    # the omap on our kvstore, to populate it with whatever
+                    # may have been set pre-bootstrap.
+                    # Probably want to lock the object here too, in case we get two
+                    # simultaenous starts (and thus omap population)
+                    keys = self._db.keys()
+                    values = list(self._db[k] for k in keys)
+                    if keys and not has_aquarium_pool:
+                        with rados.WriteOpCtx() as op:
+                            logger.info(
+                                f"Pushing {keys} to newly created aquarium pool"
+                            )
+                            self._ioctx.set_omap(op, keys, values)  # type: ignore
+                            self._ioctx.operate_write_op(op, "kvstore")
+                    # TODO: need timeouts, error handlers etc on watch
+                    self._config_watch = self._ioctx.watch(
+                        "kvstore", self._config_notify
+                    )
+                    logger.debug(
+                        f"config watch id is {self._config_watch.get_id()}"
+                    )
+                    # will raise:
+                    # rados.ObjectNotFound: [errno 2] RADOS object not found (watch error)
+            except Exception as e:
+                # TODO: deal with this (log it? ignore it?)
+                # e.g. RADOS rados state (You cannot perform that operation on a Rados object in state configuring.)
+                # this makes the log pretty messy prior to bootstrap (errors every 10 seconds)
+                logger.error(str(e))
```

Comment:
Maybe handle exceptions on a per-operation basis and log those that make sense? Propagate when the operation should actually fail?